src/components/RegisterPage.tsx

```diff
@@ -39,6 +40,10 @@ const RegisterPage = () => {
   const dispatch = useDispatch();
   const navigate = useNavigate();
 
+  const userId = useSelector((state: RootState) => {
```

Comment:
in many places you use useSelector && you also use redux toolkit
https://redux-toolkit.js.org/tutorials/typescript#project-setup
you may set useAppSelector -> thanks to that you won't need to set `RootState` anymore. which is really handy :)